src/AppBundle/Controller/HomepageController.php

```diff
@@ -2,30 +2,39 @@
 namespace AppBundle\Controller;
 use AppBundle\Entity\Category;
 use AppBundle\Entity\Product;
+use Doctrine\ORM\EntityManager;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
 use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
-use Symfony\Bundle\FrameworkBundle\Controller\Controller;
-use Symfony\Component\Routing\Annotation\Route;
 
 /**
  * @author Jan Klat <jenik@klatys.cz>
+ * @Route(service="app.controller.homepage_controller")
```

Comment:
Tomuto riadku nerozumiem...
1. V app/config/services.yml je definované, že pre service app.controller.homepage_controller je trieda AppBundle\Controller\HomepageController, tak mi príde táto definícia ako duplicitná.
2. Routing som vždy v PHP aplikáciách chápal iba z pohľadu URL. V tomto prípade to chápem iba ako pomenovanie, resp. pridelenie ID pre service aby bolo možné na ňu odkazovať. Nevidím tu však priamy súvis s URL, preto nie som si istý, či tomu správne rozumiem :-)